src/components/common/NetworkChangeOverlay/index.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import Card from "@material-ui/core/Card";
+import CardHeader from "@material-ui/core/CardHeader";
+import CardContent from "@material-ui/core/CardContent";
+import Modal from "@material-ui/core/Modal";
+import Divider from "@material-ui/core/Divider";
+import { withStyles } from "@material-ui/styles";
+
+import AlertBox, { alertTypes } from "../AlertBox";
+import { useStyles } from "./styles";
+import { walletTypes } from "../../../Redux/actionCreators/UserActions";
+
+const accountChangeAlert = {
+  header: "Incorrect Metamask Account",
+  type: alertTypes.WARNING,
+  message: `Kindly check the Address which you have set on Metamask. 
+  Please switch it to the registered address to continue using the services.`,
+};
+
+const networkChangeAlert = {
+  header: "Incorrect Metamask Network",
+  type: alertTypes.WARNING,
+  message: `Kindly check the network which you have set on Metamask. 
+  Please switch it to Mainnet to continue using the services.`,
+};
+
+class NetworkChangeOverlay extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { alert: {}, ...this.showMetaMaskConfigMismatchOverlay() };
+  }
+
+  componentDidMount() {
+    window.addEventListener("snetMMAccountChanged", this.handleMetaMaskAccountChange);
+    window.addEventListener("snetMMNetworkChanged", this.handleMetaMaskNetworkChange);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener("snetMMAccountChanged", this.handleMetaMaskAccountChange);
+    document.removeEventListener("snetMMNetworkChanged", this.handleMetaMaskNetworkChange);
+  }
+
+  showMetaMaskConfigMismatchOverlay = () => {
+    const { wallet } = this.props;
+    if (wallet && wallet.type !== walletTypes.METAMASK) {
+      return false;
+    }
+
+    const web3Provider = window.ethereum;
+    if (!web3Provider) {
+      return false;
```

Comment:
You need to return consistent structure data structure for a function.

Reply:
https://github.com/singnet/refactored-dapp/pull/184/commits/a9259c45e3e1e01c6d7b9d50299f88607e632546